### PR TITLE
Add snapshot meta

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR adds the snapshot meta data, which is needed by https://github.com/pingcap/tikv/pull/1618

PTAL @zhangjinpeng1987 @siddontang @BusyJay

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -31,6 +41,7 @@ message RaftSnapshotData {
     optional metapb.Region region   = 1;
     optional uint64 file_size       = 2;
     repeated KeyValue data          = 3;
+    optional uint64 version         = 4;
```
Comment (siddontang) [inline proto/raft_serverpb.proto]: I prefer embed the meta here directly, no need to save in data with a "snapshot meta" key.

File: proto/raft_serverpb.proto [hunk 2]
```diff
@@ -22,6 +22,16 @@ message RaftTruncatedState {
     optional uint64 term     = 2;
 }
 
+message SnapshotCFFile {
+    optional bytes cf        = 1;
```
Comment (siddontang) [inline proto/raft_serverpb.proto]: use string type

**Comment by siddontang** [comment]
Rest LGTM

**Pull request merged**